packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -246,12 +258,23 @@ class BuildInfo {
     };
   }
 
+  Map<String, dynamic>? defineConfigJsonRawMap() {
+    if (defineConfigJsonMap != null) {
+      return defineConfigJsonMap;
+    }
+    return <String, dynamic>{};
+  }
+
   /// Convert to a structured string encoded structure appropriate for usage as
   /// environment variables or to embed in other scripts.
   ///
   /// Fields that are `null` are excluded from this configuration.
   Map<String, String> toEnvironmentConfig() {
-    return <String, String>{
+    final Map<String, String> map = <String, String>{};
+    defineConfigJsonRawMap()!.forEach((String key, dynamic value) {
```

Comment:
Might want to be consistent on how we call this method, the current definition of the method doesn't allow a null to be returned

Reply:
Using filed experience, method is not necessary